internal/monitor/agent_client_test.go

```
@@ -0,0 +1,264 @@
+	}
+	if !body.closed {
+		t.Error("response body was not closed")
+	}
+}
+
+// blockedServer returns an agent that answers no request until the test ends.
+func blockedServer(t *testing.T) *httptest.Server {
+	t.Helper()
+
+	blocked := make(chan struct{})
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

golangci-lint (2.13.2)

[warning] unused-parameter: parameter 'w' seems to be unused, consider removing or renaming it as _

(revive)